src/main/java/com/adepuu/exercises/session7/User.java

```diff
@@ -0,0 +1,111 @@
+package com.adepuu.exercises.session7;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.UUID;
+
+public class User {
+  private String userId;
+
+  private String username;
+
+  private String password;
+
+  private boolean isLogin;
+
+  private List<HashMap<String, String>> users = new ArrayList<>();
+
+  public User() {
+    this.userId = UUID.randomUUID().toString();
+    this.isLogin = false;
+  }
+
+  public String getUserId() {
+    return userId;
+  }
+
+  public String getUsername() {
+    return username;
+  }
+
+  public String getPassword() {
+    return getPassword();
+  }
+
+  public boolean getIsLogin() {
+    return isLogin;
+  }
+
+  public void setIsLogin(boolean val) {
+    isLogin = val;
+  }
+
+  public void setUserName(String newUsername) {
+    username = newUsername;
+  }
+
+  public void setPassword(String newPassword) {
+    password = newPassword;
+  }
+
+  public boolean isUsernameAvailable(String username) {
+    for (HashMap<String, String> map : users) {
+      if (map.get("username").equals(username)) {
+        return false;
+      }
+    }
```

Comment:
this kind of redundant, if we are not using list, we can do it on a constant time